lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -634,6 +633,9 @@ def as_promql_label_dict(self):
             "juju_{}".format(key): val
             for key, val in self.as_dict(rename_keys={"charm_name": "charm"}).items()
         }
+        # unit should never be specified in alert rules as the apply to the entire application
```

Comment:
Typo in the comment?
Can you elaborate more?

Reply:
The problem is that by setting the unit in the alert rule label, we are also making the PromQL transformer add it to the filters it injects in the expression. Since the leader sets the alerts in the relation, the unit marked in the alert rules with the current logic is always the leader's. And since we filter the expression in the alert rules using the `juju_unit="<leader_unit_name>"`, the alert rules are evaluated ONLY for the leader.

Reply:
@dstathis, can you please incorporate the rationale Michele explains above into the comment? I think we'll want to be able to reference that later.